dietpi/dietpi-software

```diff
@@ -6427,6 +6435,52 @@ _EOF_
 			Download_Test_Media
 		fi
 
+		if To_Install 212 kavita # Kavita
+		then
+			case $G_HW_ARCH in
+				1) local arch='arm';;
+				2) local arch='arm';;
+				3) local arch='arm64';;
+				*) local arch='x64';;
+			esac
+
+			local fallback_url="https://github.com/Kareadita/Kavita/releases/download/v0.7.8/kavita-linux-$arch.tar.gz"
+			Download_Install "$(curl -sSfL 'https://api.github.com/repos/Kareadita/Kavita/releases/latest' | mawk -F\" "/\"browser_download_url\": .*\/kavita-linux-$arch\.tar\.gz\"/{print \$4}")" /opt
```

Comment:
On reinstall, `/opt/Kavita` can be removed, as configs are stored in `/mnt/dietpi_userdata/kavita`, right? To have it as failsafe as possible, I'd suggest:
1. `Download_Install` to working dir (just skip the `/opt` argument)
2. Remove `/opt/Kavita` if it exists already.
3. Move the new dir in place.

Ah, and I'd prefer `/opt/kavita` to have everything consistently in lower case. Can be renamed when moving the downloaded+extracted dir in place.

Reply:
My apologies, after checking, it seems that the config was stored inside `/opt/Kavita/config` folder. So I will change on how install, reinstall, and uninstall works:

**On install/reinstall:**
1. Create `/opt/kavita` if it's not exists.
2. Remove everything inside `/opt/kavita` except `config` folder.
3. Use `Download_Install` without `/opt` argument
4. Move the extracted files to `/opt/kavita`. Change default port to 2036 (free).
5. Create user _kavita_.
6. Create data folder (ebooks and comics)
7. Set folder/file permissions for `/opt/kavita`, ebooks and comics.
8. Setup Kavita service.
9. Finish.

**On uninstall:**
1. Remove Kavita service.
2. Remove everything inside `/opt/kavita`
3. Finish.